drools-game-engine-horserace-kjar/src/main/resources/rules/game/gamerules.drl

```diff
@@ -0,0 +1,40 @@
+package org.drools.game.horserace;
+import org.drools.game.core.api.*;
+import org.drools.game.model.api.*;
+import org.drools.game.horserace.model.*;
+
+global GameMessageService messageService;
+global GameCallbackService callback;
+
+rule "Greeter"
+    when
+        $player : Player()
+    then
+        insert(  messageService.newGameMessage(  "Welcome, " + $player.getName() + ". Venture forth to the stables and mount a mighty steed!" ) );
+end
+
+rule "start the race"
+        when
+            $player : Player()
+            $checkpoint : Checkpoint(id == "StartFinish" && !(required.playersCrossedCheckpoint contains $player.getName()))
```

Comment:
Now this rule can use the order (let's say that order = 0, is the first and last checkpoint in a race)